src/cpp/session/SessionMain.cpp

```diff
@@ -2400,6 +2404,9 @@ void rCleanup(bool terminatedNormally)
          if (error)
             LOG_ERROR(error);
       }
+      
+      // clean up locks
+      FileLock::cleanUp();
```

Comment:
We actually need to account for verification and testing codepaths here (which do an early exit from the process). Obviously this is where destructor based cleanup would shine, but all discussions about sketchiness of static destructors and controlling order of cleanup still apply in spite of this apparent advantage. We do an early exit at the following locations:

https://github.com/rstudio/rstudio/blob/feature/directory-file-locking/src/cpp/session/SessionMain.cpp#L1792
https://github.com/rstudio/rstudio/blob/feature/directory-file-locking/src/cpp/session/SessionMain.cpp#L1799

I'd create a new `exitEarly` helper function which takes the return status and does a FileLock::cleanup before actually calling `::exit`